db/table_cache.cc

```diff
@@ -365,17 +383,7 @@ void TableCache::CreateRowCacheKeyPrefix(const ReadOptions& options,
   // the same as there is no snapshot. The exception is that if
   // a seq-checking callback is registered, some internal keys
   // may still be filtered out.
-  uint64_t seq_no = 0;
-  // Maybe we can include the whole file ifsnapshot == fd.largest_seqno.
-  if (options.snapshot != nullptr &&
-      (get_context->has_callback() ||
-       static_cast_with_check<const SnapshotImpl>(options.snapshot)
-               ->GetSequenceNumber() <= fd.largest_seqno)) {
-    // We should consider to use options.snapshot->GetSequenceNumber()
-    // instead of GetInternalKeySeqno(k), which will make the code
-    // easier to understand.
-    seq_no = 1 + GetInternalKeySeqno(internal_key);
-  }
+  uint64_t seq_no = GetSequenceNumber(options, fd, internal_key, get_context);
```

Comment:
I think it would help make the code more readable to make this `seq_no` an output argument of `CreateRowCacheKeyPrefix` too, and use it below in the call to `replayGetContextLog` instead of calling `GetSequenceNumber` twice. WDYT?

Reply:
Since the logic of `GetSequenceNumber` is not being used in other places, probably I can simply remove this function. 

The only concern about returning the `seq_no` as a output is more relating to coding style and having unused return value. But I think we are good to return `seq_no` as this `CreateRowCacheKeyPrefix` is only used in two places.